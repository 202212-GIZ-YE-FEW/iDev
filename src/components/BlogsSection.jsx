import { withTranslation } from "next-i18next";
import Carousel from "react-multi-carousel";

import "react-multi-carousel/lib/styles.css";

import BlogItem from "@/components/BlogItem";
const BlogsSection = ({
    t,
    textSize = "text-3xl md:text-4xl rtl:md:text-3xl lg:text-5xl rtl:lg:text-4xl",
    title = "recentBlogsTitle",
    blogs = [],
}) => {
    const responsive = {
        superLargeDesktop: {
            breakpoint: { max: 4000, min: 3000 },
            items: 5,
        },
        desktop: {
            breakpoint: { max: 3000, min: 1024 },
            items: 3,
        },
        tablet: {
            breakpoint: { max: 1024, min: 464 },
            items: 2,
        },
        mobile: {
            breakpoint: { max: 464, min: 0 },
            items: 1,
        },
    };
    return (
        <>
            <div className='flex flex-col space-y-16 justify-between'>
                <p
                    className={
                        textSize + "font-normal block uppercase break-words"
                    }
                >
                    {t(`${title}`)}
                </p>
                <Carousel infinite='true' responsive={responsive}>
                    {blogs?.map((item, index) => {
                        return (
                            <BlogItem
                                key={index}
                                ar_title={item.body.ar_title}
                                en_title={item.body.en_title}
                                id={item.id}
                                isOdd={index % 2 === 0 ? false : true}
                            />
                        );
                    })}
                </Carousel>
            </div>
        </>
    );
};
export default withTranslation("blog")(BlogsSection);
