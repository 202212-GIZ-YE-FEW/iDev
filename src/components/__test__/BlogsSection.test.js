import renderer from "react-test-renderer";

import BlogsSection from "../BlogsSection";
it("renders correctly", () => {
    const tree = renderer.create(<BlogsSection />).toJSON();
    expect(tree).toMatchSnapshot();
});
