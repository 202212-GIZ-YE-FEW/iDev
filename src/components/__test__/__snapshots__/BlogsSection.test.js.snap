// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex flex-col space-y-16 justify-between"
>
  <p
    className="text-3xl md:text-4xl rtl:md:text-3xl lg:text-5xl rtl:lg:text-4xlfont-normal block uppercase break-words"
  >
    common:recentBlogsTitle
  </p>
  <div
    className="react-multi-carousel-list  "
    dir="ltr"
  >
    <ul
      className="react-multi-carousel-track "
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "overflow": "unset",
          "transform": "translate3d(0px,0,0)",
          "transition": "none",
        }
      }
    />
  </div>
</div>
`;
